src/settings/util/validation.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+  FormattedMessage,
+} from 'react-intl';
+
+import {
+  FILTERING_CONDITIONS_FIELDS,
+  EMPTY_VALIDATION_STATE_FOR_FIELD,
+} from '../constants';
+
+export const isFilteringConditionsFilled = (filteringConditions) => (
+  !!filteringConditions.filter((filteringCondition) => (
+    filteringCondition[FILTERING_CONDITIONS_FIELDS.NAME] &&
+    filteringCondition[FILTERING_CONDITIONS_FIELDS.ACTIVE] &&
+    filteringCondition[FILTERING_CONDITIONS_FIELDS.SET_SPEC] &&
+    filteringCondition[FILTERING_CONDITIONS_FIELDS.VALUE]
+  )).length
+);
+
+export const validateRequiredField = (value) => (
+  value ? EMPTY_VALIDATION_STATE_FOR_FIELD : <FormattedMessage id="ui-oai-pmh.settings.sets.edit.validationMessage.required" />
+);
+
+export const validateFieldLength = (value, maxLength) => {
+  if (value) {
+    const str = value.toString();
+    const errorMsg = (
+      <FormattedMessage
+        id="ui-oai-pmh.settings.sets.edit.validationMessage.maxLength"
+        values={{
+          maxLength,
+        }}
+      />
+    );
+
+    return str.length <= maxLength ? EMPTY_VALIDATION_STATE_FOR_FIELD : errorMsg;
+  }
+
+  return EMPTY_VALIDATION_STATE_FOR_FIELD;
```

Comment:
I would suggest dealing with error message stuff only when needed.

```suggestion
  const str = value.toString();

  if (value && str.length > maxLength) {
    return (
      <FormattedMessage
        id="ui-oai-pmh.settings.sets.edit.validationMessage.maxLength"
        values={{
          maxLength,
        }}
      />
    );
  }

  return EMPTY_VALIDATION_STATE_FOR_FIELD;
```

Reply:
Fixed